trio/_util.py

```diff
@@ -338,7 +354,7 @@ def _create(cls: t.Type[T], *args: object, **kwargs: object) -> T:
         return super().__call__(*args, **kwargs)  # type: ignore
 
 
-def name_asyncgen(agen):
+def name_asyncgen(agen: t.Any) -> str:
```

Comment:
any reason this is left as `Any`?

Reply:
The `hasattr` check made me expect that this was intended to be callable on any object, with a graceful fallback. But looking at the call site, it's only going to be called via `sys.set_asyncgen_hooks()`. So it should always be a real async gen object, or something that's duck typing very well (Cython).

Reply:
the docstring looks to imply it should only be called with AsyncGen iterators.